src/lib/CoreInstance.cc

```diff
@@ -175,6 +178,9 @@ void CoreInstance::createL1DataMemory(const simeng::MemInterfaceType type) {
 
 void CoreInstance::setL1DataMemory(
     std::shared_ptr<simeng::MemoryInterface> memRef) {
+  assert(setDataMemory_ &&
```

Comment:
For some reason this assertion is not firing in my SST Build, I changed the function to:
```
void CoreInstance::setL1DataMemory(
    std::shared_ptr<simeng::MemoryInterface> memRef) {
  std::cout << "Set Data Memory: " << setDataMemory_ << std::endl;
  assert(setDataMemory_ &&
         "setL1DataMemory(...) called but the interface was created by the "
         "CoreInstance class.");
  // Set the L1D cache instance to use
  dataMemory_ = memRef;
  return;
}
```
The output I get is below and the simulation still continues. Can you please confirm that this happens to you as well.
![image](https://user-images.githubusercontent.com/62028695/191285732-bcdf1845-ebe7-426a-a438-78e17208b319.png)

I encountered this because the config.py in the sst folder still has the path to `a64fx.yaml` and not to `a64fx-sst.yaml`. So the assertion should've failed however it doesn't.

Reply:
You've built with release mode. No assertions will trigger. Please rebuild and re-try.